### ISSUE TITLE
test_rnn_clone is flaky when all seed-setting code is removed

### ISSUE THREAD

**Issue by melonwater211** [learnables/learn2learn#251]
### Introduction
`test_rnn_clone` in `tests/unit/utils_test.py` seems to be flaky when all seed-setting code (e.g. `np.random.seed(0)` or `tf.random.set_seed(0)`) is commented out. 

In commit 6b741028f0812ba73da58acf7206d93654dde97a, `test_rnn_clone` failed ~5% of the time (out of 500 runs) compared to 0% of the time (out of 500 runs) when no seed-setting code is removed.   

### Motivation
Some tests can be flaky with high failure rates, but are not discovered when the seeds are set. We are trying to stabilize such tests.

### Environment
The tests were run using `pytest 6.2.2` in a `conda` environment with `Python 3.6.13`. The OS used was `Ubuntu 16.04`. 

### Solutions
One possible solution to reduce flakiness is to change the parameters used for prediction. We tried changing the following parameters.

Increasing `N` to 6 and 10 increased flakiness to ~20% and ~28% respectively. 

*These changes increased runtimes by ~40% and ~80% respectively.*

Decreasing `N` to 2 decreased flakiness to ~4%. 

*This change did not increase runtimes significantly.*

Decreasing the `lr` parameter in the `l2l.algorithms.MAML()` call to 1e-4 decreased flakiness to ~1%.

Decreasing the `lr` parameter in the `torch.optim.SGD()` call to 0.0001 decreased flakiness to ~1%. 

*Theses changes increased runtimes by ~10%.*

Please let us know if there are any other solutions that should be incorporated. We will be happy to raise a Pull Request to fix the tests and incorporate any feedback that you may have.

**Comment by seba-1511** [comment]
Hello @melonwater211 ,

Thanks for checking our RNN tests. Those tests check if the RNN is able to improve with MAML's objective, which is not always the case with random data (requires careful tuning of the learning rate, hence the fixed seed).  If you have a solution that works 100% of the time when seeding is removed, I'd be happy to merge it. Otherwise I'd rather keep the imperfect but always passing test (it is quite randomly passing tests are very annoying with CI).

**Comment by melonwater211** [comment]
Hi @seba-1511.

After additional testing, we have determined that reducing both the `lr` parameter in the `l2l.algorithms.MAML()` call to 0.00010371638283999379 and the `lr` parameter in the `torch.optim.SGD()` call to 0.00037820735073939584 will reduce the flakiness of `test_rnn_clone ` to 0.09%. In other words, the test will fail only 9 out of every 10000 runs. Please let us know if this solution would work for you. Thank you.
